src/v/cluster/tx_gateway.cc

```diff
@@ -56,24 +57,32 @@ tx_gateway::abort_tx(abort_tx_request&&, rpc::streaming_context&) {
     return ss::make_ready_future<abort_tx_reply>(abort_tx_reply());
 }
 
-ss::future<begin_group_tx_reply>
-tx_gateway::begin_group_tx(begin_group_tx_request&&, rpc::streaming_context&) {
-    co_return begin_group_tx_reply();
+ss::future<begin_group_tx_reply> tx_gateway::begin_group_tx(
+  begin_group_tx_request&& request, rpc::streaming_context&) {
+    return _rm_group_frontend.local().do_begin_group_tx(
```

Comment:
did you mean to call `prepare_group_tx` which would repeat the check of the local node being a leader and re-dispatch locally or remotely if needed. this here looks like it assumes we're on the correct node and calls the local group router directly. same question for the rest of these rpc handlers.

Reply:
No, I intented to call `do_` method. `tx_gateway_frontend` invokes `rm_group_frontend::prepare_group_tx` which may dispatch a call to a remote node using `tx_gateway` so when it processes a request it uses `do_prepare_group_tx` because it already should be on the right node.

If `tx_gateway` uses `prepare_group_tx` it may create a infinite bouncing rpc loop when metadata happen to be stale.